dev/ProgressRing/ProgressRing.xaml

```diff
@@ -21,29 +21,55 @@
+                                                Storyboard.TargetProperty="Opacity"
+                                                To="0"
+                                                Duration="0:0:0.15" />
+                                        </Storyboard>
+                                    </VisualTransition>
+                                </VisualStateGroup.Transitions>
+                                <VisualState x:Name="Inactive">
                                     <Storyboard>
                                         <DoubleAnimation
                                             Storyboard.TargetName="LayoutRoot"
                                             Storyboard.TargetProperty="Opacity"
-                                            To="1"
-                                            Duration="0:0:0.15" />
+                                            To="0"
+                                            Duration="0" />
                                     </Storyboard>
                                 </VisualState>
-                                <VisualState x:Name="Inactive">
+                                <VisualState x:Name="DeterminateActive">
                                     <Storyboard>
                                         <DoubleAnimation
-                                            Storyboard.TargetName="LayoutRoot"
+                                            Storyboard.TargetName="DeterminateLottiePlayer"
+                                            Storyboard.TargetProperty="Opacity"
+                                            To="1"
+                                            Duration="0" />
+                                        <DoubleAnimation
+                                            Storyboard.TargetName="IndeterminateLottiePlayer"
                                             Storyboard.TargetProperty="Opacity"
                                             To="0"
                                             Duration="0" />
                                     </Storyboard>
                                 </VisualState>
+                                <VisualState x:Name="IndeterminateActive" />
+
                             </VisualStateGroup>
                         </VisualStateManager.VisualStateGroups>
-                        <!-- AnimatedVisualPlayer for Lottie -->
-                        <local:AnimatedVisualPlayer x:Name="IndeterminateAnimatedVisualPlayer" AutoPlay="false" Stretch="fill"/>
 
+                        <!-- Determinate AnimatedVisualPlayer for Lottie -->
+                        <local:AnimatedVisualPlayer x:Name="DeterminateLottiePlayer" AutoPlay="false" Stretch="fill" Opacity="0"/>
+
+                        <!-- Indeterminate AnimatedVisualPlayer for Lottie -->
+                        <local:AnimatedVisualPlayer x:Name="IndeterminateLottiePlayer" AutoPlay="false" Stretch="fill"/>
```

Comment:
>AnimatedVisualPlayer [](start = 31, length = 20)

I thought that the goal was to use the same VisualPlayer for both, just swaping the source. Were we upable to get the transition animations to work?

Reply:
I originally refactored it to have only one animated visual player (much like the ProgressUI prototype), only to realize that swapping `Source()` with code behind doesn't allow re-templating. We can discuss this more? + @ranjeshj